tests/system/source/main.d

```diff
@@ -38,127 +38,93 @@ import core.time;
+            ulong node_height;
+            Duration duration = 0.seconds;
+            do
             {
-                writefln("%s Client #%s is at block height %s", PREFIX, idx, getHeight);
-                const blocks = client.getBlocksFrom(0, 42);
-                writefln("%s Client #%s has blocks:\n%s", PREFIX, idx, prettify(blocks));
-                writefln("%s ----------------------------------------", PREFIX);
-                assert(blocks.length == height + 1);
-                if (idx != 0)
-                    assert(blockHash == hashFull(blocks[height].header));
+                node_height = client.getBlockHeight();
+                if (node_height < height) // Only sleep when we need to
+                {
+                    writefln("%s %s has height %s not yet height %s: sleep %s (so far %s out of %s max) ",
+                        PREFIX, nodeFromClientIndex(idx), node_height, height, retry_delay,
+                        duration == 0.seconds ? "0 secs" : duration.toString(), max_duration);
+                    Thread.sleep(retry_delay);
+                }
                 else
-                    blockHash = hashFull(blocks[height].header);
+                {
+                    writefln("%s %s is at block height %s", PREFIX, nodeFromClientIndex(idx), node_height);
+                    const blocks = client.getBlocksFrom(node_height + 1, 10);
+                    writefln("%s %s has blocks:\n%s", PREFIX, nodeFromClientIndex(idx), prettify(blocks));
+                    writefln("%s ----------------------------------------", PREFIX);
+                }
+                duration += retry_delay;
             }
+            while (node_height < height && duration < max_duration); // Retry if we're too early
+            assert(node_height >= height,
+                format!"%s %s still has block height %s less than expected height %s after more than %s"
+                    (PREFIX, nodeFromClientIndex(idx), node_height, height, duration));
         }
-        while (getHeight < height && times++ < 50); // Retry if we're too early
-        assert(getHeight == height,
-            format!"%s Client #%s still has block height %s not %s"
-                (PREFIX, idx, getHeight, height));
-        times = 0;
     }
+
+    clients.enumerate.each!((idx, client) =>
+    {
+        writefln("%s %s info: %s", PREFIX, nodeFromClientIndex(idx), client.getNodeInfo());
+        const height = client.getBlockHeight();
+        writefln("%s %s has block height %s", PREFIX, nodeFromClientIndex(idx), height);
+        writefln("%s ----------------------------------------", PREFIX);
+    }());
+
+    // Make sure the nodes use the test genesis block
+    const blocks = clients[0].getBlocksFrom(0, 1);
+    assert(blocks.length == 1);
+    assert(blocks[0] == TestGenesis.GenesisBlock, format!"%s Not using expected TestGenesis.GenesisBlock"(PREFIX));
```

Comment:
I don't think we need this anymore, since we're not making txs ourselves. But it doesn't hurt to keep it either.

Reply:
In faucet we assume agora is using this GenesisBlock so I think it is worth keeping.